### PR TITLE
Setup code quality tools

### PR DESCRIPTION
**Pull request by t-romani**
Issue #2

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    - http
+    - https
+
+Metrics/ParameterLists:
+  Max: 5
+  CountKeywordArgs: true
+
+Metrics/PerceivedComplexity:
+  Max: 12
+
```
Comment (brunvez) [inline .rubocop.yml]: I generally prefer the developer to make these decisions, maybe we can just warn or remove it. I don't know just an opinion

File: .reek.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    - "/[A-Z]/"
+    accept:
+    - _
+
+  UnusedParameters:
+    enabled: true
+    exclude: []
+
+  UnusedPrivateMethod:
+    enabled: false
```
Comment (brunvez) [inline .reek.yml]: These could be useful

File: .rubocop.yml (hunk 1 above)
Comment (t-romani) [reply .rubocop.yml]: Maybe we could set some of those as warnings, let's discuss this better.

File: .github/database.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+default: &default
```
Comment (brunvez) [inline .github/database.yml]: This is no longer needed

File: .codeclimate.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    config:
+      threshold: 4
+  similar-code:
+    enabled: true
+    config:
+      threshold: #language-specific defaults. overrides affect all languages.
+  identical-code:
+    enabled: true
+    config:
+      threshold: #language-specific defaults. overrides affect all languages.
```
Comment (brunvez) [inline .codeclimate.yml]: We can just skip these ones

File: .codeclimate.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    config:
+      threshold: 250
+  method-complexity:
+    enabled: true
+    config:
+      threshold: 5
+  method-count:
+    enabled: true
+    config:
+      threshold: 20
```
Comment (brunvez) [inline .codeclimate.yml]: Same here

File: .codeclimate.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    config:
+      threshold: 4
+  complex-logic:
+    enabled: true
+    config:
+      threshold: 4
+  file-lines:
+    enabled: true
+    config:
+      threshold: 250
```
Comment (brunvez) [inline .codeclimate.yml]: We got rid of this on the other tools, we can get rid of it here too

File: .codeclimate.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  method-complexity:
+    enabled: true
+    config:
+      threshold: 5
```
Comment (brunvez) [inline .codeclimate.yml]: Same here

File: .codeclimate.yml [hunk 8]
```diff
@@ -0,0 +1,73 @@
+version: '2'
+checks:
+  argument-count:
+    enabled: true
+    config:
+      threshold: 4
```
Comment (brunvez) [inline .codeclimate.yml]: We can do without this

File: .reek.yml [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+  TooManyInstanceVariables:
+    enabled: true
+    exclude: []
+    max_instance_variables: 9
+
+  TooManyMethods:
+    enabled: true
+    exclude: []
+    max_methods: 25
```
Comment (brunvez) [inline .reek.yml]: Nope

File: .reek.yml [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+  TooManyStatements:
+    enabled: true
+    exclude:
+    - initialize
+    max_statements: 12
```
Comment (brunvez) [inline .reek.yml]: Nope

File: .rubocop.yml [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+Lint/AmbiguousBlockAssociation:
+  Exclude:
+    - spec/**/*
+
+Metrics/AbcSize:
+  Enabled: false
+  # The ABC size is a calculated magnitude, so this number can be an Integer or
+  # a Float.
+  Max: 18
```
Comment (brunvez) [inline .rubocop.yml]: Nope

File: .rubocop.yml [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+Metrics/MethodLength:
+  CountComments: false  # count full line comments?
+  Max: 24
+  Severity: warning
+
+Metrics/ModuleLength:
+  CountComments: false  # count full line comments?
+  Max: 200
+  Enabled: false
```
Comment (brunvez) [inline .rubocop.yml]: I thought we were getting rid of these

File: .rubocop.yml [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    - https
+
+Metrics/ParameterLists:
+  Max: 5
+  CountKeywordArgs: true
+  Enabled: false
+
+Metrics/PerceivedComplexity:
+  Max: 12
+  Enabled: false
```
Comment (brunvez) [inline .rubocop.yml]: Same

**Pull request merged**